Update role defaults like molecule_docker_ci

tehsmyers opened the pull request:
These updates are inspired by simplifications and improvements seen in
the molecule_docker_ci OASIS role, such as defaulting to the 'present'
state, and providing a reasonable default cloud name instead of default
to `omit`.

README examples were also updated with the more concise role invocation
found in molecule_docker_ci's README.

tehsmyers commented on README.md:
```diff
@@ -42,9 +42,10 @@ Currently the following variables are supported:
-* `molecule_openstack_ci_cloud` - name of the OpenStack cloud to connect to. This defaults to being omitted,
- in which case the openstack libraries will read the environment variable "OS\_CLOUD" by default. Set this
- value if you want to override that for specific cases or do not have the value set.
+* `molecule_openstack_ci_cloud` - name of the OpenStack cloud to connect to. This defaults to the value
```
This, by the way, is how I actually ended up here. Despite having `OS_CLOUD` set, os_server (and its underlying machinery) was not detecting the default cloud, and it seemed odd to be carrying around the `{{ lookup('env', 'OS_CLOUD') | default('default', true) }}` from role to role. The simpler and more user-friendly approach taken by `molecule_docker_ci` was an inspiration for making this simpler.

greg-hellings commented:
Should the molecule scenario inside of this role be updated to reflect this change? I'm not sure it needs to be, but it doesn't hurt anything to keep it at the most streamlined version of itself that it can be.

tehsmyers commented:
I did do that, but it looks like I may have forgotten to commit the changes. I'll get that taken care of.

Commit:
Update role defaults like molecule_docker_ci

These updates are inspired by simplifications and improvements seen in
the molecule_docker_ci OASIS role, such as defaulting to the 'present'
state, and providing a reasonable default cloud name instead of default
to `omit`.

README examples were also updated with the more concise role invocation
found in molecule_docker_ci's README.

Commit:
Update travis config with current standards

greg-hellings reviewed (approved):
Yes, please.
